doc/modules/prediction.rst

```diff
@@ -0,0 +1,156 @@
+.. currentmodule:: sklearn.model_selection
+
+.. _cutoffclassifier:
+
+========================================================
+Tuning cut-off decision threshold for classes prediction
+========================================================
+
+Classifiers are predictive models: they use statistical learning to predict
+outcomes. The outcomes of a classifier takes two forms: a "soft" score for each
+sample in relation to each class, and a "hard" categorical prediction (i.e.
+class label). Soft predictions are obtained using :term:`predict_proba` or
+:term:`decision_function` while hard predictions are obtained using
+:term:`predict`.
+
+In scikit-learn, there is a connection between soft and hard prediction. In the
+case of a binary classification, hard predictions are obtained by associating
+the positive class with probability value greater than 0.5 (obtained with
+:term:`predict_proba`) or decision function value greater than 0 (obtained with
+:term:`decision_function`).
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.tree import DecisionTreeClassifier
+    >>> X, y = make_classification(random_state=0)
+    >>> classifier = DecisionTreeClassifier(max_depth=2, random_state=0).fit(X, y)
+    >>> classifier.predict_proba(X[:4])
+    array([[0.94   , 0.06   ],
+           [0.94   , 0.06   ],
+           [0.04..., 0.95...],
+           [0.04..., 0.95...]])
+    >>> classifier.predict(X[:4])
+    array([0, 0, 1, 1])
+
+
+Similar rules apply for other classification problems.
+
+While these approaches are reasonable as default behaviors, they might not be
+adapted to some cases. The context and nature of the use case define the
+expected behavior of the classifier and thus the strategy to convert soft
+predictions into hard predictions. We illustrate this point with an example.
+
+Let's imagine the deployment of a predictive model helping medical doctors to
+detect cancers. In a setting where this model was a tool to discard obvious
+cases and false positives don't lead to potentially harmful treatments, doctors
+might be interested in having a high recall (all cancers cases should be tagged
+as such) to not miss any patient with a cancer. However, that is at the cost of
+having more false positive predictions (i.e. lower precision). Thus, in terms of
+decision threshold, it may be better to classify a patient as having a cancer
+for a probability lower than 0.5.
```

Comment:
nit: should we call the output of `predict_proba` a probability? I thought (but can't point you to a reference) that except for some few models the score that a model outputs is more like a "the higher the more likely it is the positive class" but that you should not interpret it as a probability. Or said differently, if you want to interpret it as a probability you need to calibrate it.

Reply:
`predict_proba` is a predicted (conditional) probability. The question how good it is is a  different one. For that, calibration is very important. Good models don't need to be calibrated, they already are like a well/correctly specified logistic regression.